File: backend/app/services/embeddings.py
from openai import OpenAI
from app.core.config import settings
from typing import List

class EmbeddingService:
    def __init__(self):
        self.client = OpenAI(api_key=settings.OPENAI_API_KEY) if settings.OPENAI_API_KEY else None

    def get_embeddings(self, texts: List[str]) -> List[List[float]]:
        """
        Generates embeddings for a list of texts using OpenAI.
        """
        if not self.client:
            # Fallback or error
            return []
            
        response = self.client.embeddings.create(
            input=texts,
            model="text-embedding-3-small"
        )
        return [data.embedding for data in response.data]

embedding_service = EmbeddingService()

File: backend/services/embeddings.js
const OpenAI = require("openai");
require('dotenv').config();

class EmbeddingService {
  constructor() {
    this.client = process.env.OPENAI_API_KEY ? new OpenAI({ apiKey: process.env.OPENAI_API_KEY }) : null;
  }

  async getEmbeddings(texts) {
    if (!this.client || !texts || texts.length === 0) {
      return [];
    }

    try {
      const response = await this.client.embeddings.create({
        input: texts,
        model: "text-embedding-3-small",
      });
      return response.data.map(item => item.embedding);
    } catch (error) {
      console.error("Error in OpenAI embeddings:", error);
      return [];
    }
  }
}

module.exports = new EmbeddingService();
